OWML.ModHelper.Menus/ModInputCombinationElement.cs

```diff
@@ -0,0 +1,171 @@
+
+        public ILayoutManager Layout { get; private set; }
+
+        override public string Title
+        {
+            get => _combination;
+            set
+            {
+                _combination = value;
+                UpdateContents();
+            }
+        }
+
+        private string _combination;
+        private readonly GameObject _layoutObject;
+        private readonly IModInputHandler _inputHandler;
+
+        private static IModInputCombinationElementMenu _popupMenu;
+
+        public ModInputCombinationElement(TwoButtonToggleElement toggle, IModMenu menu, IModInputCombinationElementMenu popupMenu, IModInputHandler inputHandler, string combination = "") : base(toggle, menu)
+        {
+            _inputHandler = inputHandler;
+            _combination = combination;
+            _layoutObject = toggle.transform.GetChild(1).GetChild(0).GetChild(1).gameObject;
+            var layoutGroup = _layoutObject.GetComponent<HorizontalLayoutGroup>();
+            var scale = toggle.transform.localScale;
+
+            var commandObject = new GameObject();
+            var commandComponent = commandObject.AddComponent<ModCommandListener>();
+            commandComponent.Initialize(InputLibrary.interact);
+            commandComponent.OnNewlyReleased += OnEditButton;
+            YesButton.Title = "Edit";
+            YesButton.OnClick += OnEditClick;
+
+            var deleteCommand = new SingleAxisCommand();
+            var deleteBindingGmpd = new InputBinding(JoystickButton.FaceUp);
+            var deleteBindingKbrd = new InputBinding(KeyCode.Delete);
+            deleteCommand.SetInputs(deleteBindingGmpd, deleteBindingKbrd);
+            commandObject = new GameObject();
+            var updater = commandObject.AddComponent<ModCommandUpdater>();
+            updater.Initialize(deleteCommand);
+            commandComponent = commandObject.AddComponent<ModCommandListener>();
+            commandComponent.Initialize(deleteCommand);
+            commandComponent.OnNewlyReleased += OnDeleteButton;
+            NoButton.Title = "Delete";
+            NoButton.OnClick += OnDeleteClick;
+
+            Initialize(menu);
+            layoutGroup.childControlWidth = false;
+            layoutGroup.childControlHeight = false;
+            layoutGroup.childForceExpandHeight = false;
+            layoutGroup.childForceExpandWidth = false;
+            layoutGroup.spacing = 0f;
+            var constantGraphics = _layoutObject.GetComponentsInChildren<Graphic>(true);
+            _layoutObject.transform.GetChild(1).gameObject.SetActive(false);
+            Layout = new LayoutManager(layoutGroup, MonoBehaviour.FindObjectOfType<UIStyleManager>(),
+                ModUIStyleApplier.ReplaceStyleApplier(toggle.gameObject), scale, constantGraphics);
+            UpdateContents();
+            _popupMenu = popupMenu;
+        }
+
+        private void UpdateContents()
+        {
+            Layout.Clear();
+            if (_combination != "")
```

Comment:
invert if?

Reply:
I need to call UpdateState regardless and don't really want to write it in two places